### DISCUSSION TITLE
Default values for configuration?

### DISCUSSION THREAD

**Discussion by rethab** [origin-energy/java-snapshot-testing#137] [General]
Hi,

would you be open for a PR to define some defaults for config values?

For example, the following three are probably safe defaults so users wouldn't need to repeat them:

```text
snapshot-dir=__snapshots__
output-dir=src/test/java
ci-env-var=CI
```

Obviously it should be built in a way that values can still be overridden via the properties file.

**Comment by jackmatt2** [comment]
I did consider defaults for some of these properties initially.  Ultimately, I decided being explicit it better for the following reasons.

1.  You only need to create the `snapshot.properties` file once.  And probably won't ever need to change it again. In most cases it's just a copy paste from the docs.
2. `output-dir`: Being explicit about the JVM language makes it clear to **Kotlin** and **Gradle** developers that it needs to be changed. If it had a default - it would be confusing for these users as to why it was outputting files into a `src/test/java` directory. They may quickly abandon it thinking these JVM languages are not supported.
3. `ci-env-var`: Although common on cloud CI environments - being explicit here makes the user aware of exactly what environment variable is being used. Perhaps it's not defined on their in house CI system.  This makes it clear what is being used - no magic or reading though source code.
4. `snapshot-dir`: Being explicit makes it clear that this can be changed if the user doesn't like this particular folder name or coding conventions won't allow it.

At the end of the day - copying 3 lines of code is the same effort as copying 10 lines.  I believe the benefits of being explicit outweigh any cons.

**Reply by rethab** [reply]
Okay, thanks for clarifying :)
